front/app/components/UI/QuillEditor/index.tsx

```diff
@@ -246,14 +246,47 @@ VideoFormat.tagName = 'iframe';
 Quill.register(VideoFormat, true);
 // END allow image & video resizing styles
 
+// BEGIN function to detect whether urls are external
+// inspired by https://github.com/quilljs/quill/blob/develop/formats/link.js#L33
+const ALLOWED_PROTOCOLS = new Set(['http', 'https', 'mailto', 'tel']);
+
+function isExternal(url: string) {
+  const protocol = url.slice(0, url.indexOf(':'));
+  return ALLOWED_PROTOCOLS.has(protocol);
+}
+// END function to detect whether urls are externel
+
+// BEGIN custom link implementation
+const Link = Quill.import('formats/link');
+
+class CustomLink extends Link {
+  static create(url) {
+    const node = super.create(url);
+
+    if (!isExternal(url)) {
+      node.removeAttribute('target');
+    }
+
```

Comment:
Perhaps you could still add a comment to mention that with links the target=_blank is added by default, which is why you remove it.

Reply:
Added a comment to explain that! Also opened a new issue on Jira for protocol problem: https://citizenlab.atlassian.net/browse/CL2-6548?atlOrigin=eyJpIjoiZTcxYmFiZjRhN2MzNGEyNWJlMmI1YjhmMTdmNTIyYTYiLCJwIjoiaiJ9. I will take care of this asap.